### PR TITLE
Fixes email send issue related to bad emails

### PR DESCRIPTION
**Pull request by alanhartless**
**Description**
See #582.  When a bad email is encountered during a batch email send, subsequent email sends failed due to Mautic retaining the previous error.  This PR fixes the issue so that the process is not stopped.

**Testing**
Create several leads.  The first, use a bad email such as `bad.@gmail.com`.  The others, use a good email.  Add the three leads to a clean lead list.  Create a new list based email and associate that list to the email.  Send to the leads.  If the bad email is encountered before the others, the rest will fail.

Apply the PR and all the good ones should still send and thus only the bad emails listed as failed.

### CODE REVIEW

**Comment by jhrocha** [comment]
Awesome. It's worked for me !

**Comment by escopecz** [comment]
I'm not able to find the problem. Is this it? It seems OK to me.
mauticbademail

**Comment by jhrocha** [comment]
The error occurs when there are correct and incorrect emails in the delivery list. Sending happens by the time a bad email is found, so the subsequent emails are not sent, even with correct address.
The fix worked for me.

File: app/bundles/EmailBundle/Entity/EmailRepository.php [hunk 1]
```diff
@@ -198,6 +198,7 @@ public function getEmailPendingLeads($emailId, $listIds = null, $countOnly = fal
                 ->setMaxResults($limit);
         }
 
+        $q->orderBy('l.id');
```
Comment (escopecz) [inline app/bundles/EmailBundle/Entity/EmailRepository.php]: This change causes following error on Postgres
```
 500 Internal Server Error - An exception occurred while executing 'SELECT count(l.id) as count FROM mau_leads l INNER JOIN mau_lead_lists_leads ll ON l.id = ll.lead_id INNER JOIN mau_email_list_xref el ON el.leadlist_id = ll.leadlist_id WHERE (el.email_id = 24) AND (l.id NOT IN (SELECT dne.lead_id FROM mau_email_donotemail dne WHERE dne.lead_id IS NOT NULL)) AND (l.id NOT IN (SELECT stat.lead_id FROM mau_email_stats stat WHERE (stat.lead_id IS NOT NULL) AND (stat.email_id = el.email_id))) AND ((l.email IS NOT NULL) OR (l.email <> '')) ORDER BY l.id ASC': SQLSTATE[42803]: Grouping error: 7 ERROR: column "l.id" must appear in the GROUP BY clause or be used in an aggregate function LINE 1: ...((l.email IS NOT NULL) OR (l.email <> '')) ORDER BY l.id ASC ^
```
If I add also `$q->groupBy('l.id');`, the query works fine.

**Comment by alanhartless** [comment]
@escopecz I added the groupBy clause.

**Comment by escopecz** [comment]
This PR fixes the issue :+1:

**Comment by MaxOnTheHill** [comment]
Hey guys .... 
looks like it checks the emails and skips bad ones so the good ones continue to send out.  This is great to get all the good email addresses sent out [ BONUS ]…. but perhaps when it finds a bad one it could flag them for correction - can’t see that it does this?  
Otherwise you have no way of knowing which leads have bad email addresses/data that need fixing.

**Pull request merged**